org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/async/StepMessageOut.scala

```diff
@@ -50,38 +59,57 @@ case class StepMessageOut(debugTarget: ScalaDebugTarget, thread: ScalaThread) ex
     thread.resumeFromScala(DebugEvent.CLIENT_REQUEST)
   }
 
+  private[async] def findThisType(stepEvent: StepEvent): String = {
+    val loc = stepEvent.location()
+    val declType = loc.declaringType().name()
+    Try {
```

Comment:
Try is so functional... 
Sometimes JDI is not too responsive and for example invalidates the stack frame/trace on debugging machine (I'm not sure if my explanation is clear). So here we could have some strategies to take:
1. allow exception to be thrown and debugging process is stopped
2. try another call with JDI and if succeeded then continue debugging otherwise rethrow and break
3. drop to some default and allow debugging, the method to stop is not found at most.
The problem for me is just to assess what fraction of JDI calls ends up with exception...